src/scripts/mint.ts

```diff
@@ -0,0 +1,305 @@
+import { BaseError, createPublicClient, http, parseUnits } from "viem";
+import { getAllCollaterals, getCollateralInformation, mintDollar } from "./faucet";
+import {
+  allowanceButton,
+  collateralInput,
+  collateralSelect,
+  dollarInput,
+  governanceCheckBox,
+  governanceFormControl,
+  governanceInput,
+  minDollarInput,
+  mintButton,
+} from "./ui";
+import { approveToSpend, getAllowance, getTokenDecimals } from "./erc20";
+import { getConnectedClient } from "./connect-wallet";
+import { diamondAddress, dollarAddress, ubqAddress } from "./constants";
+import { ToastActions } from "./toast";
+import { mainnet } from "viem/chains";
+
+let selectedCollateralIndex = 0;
+let dollarAmount = 0;
+let dollarOutMin = 0;
+let maxCollateralIn = 0;
+let maxGovernanceIn = 0;
+let isOneToOne = false;
+
+let isButtonInteractionsDisabled = false;
```

Comment:
I don't think there is a need to have this variable as you can just use mintButton.disabled to check the status. This could be the cause for the button's inconsistency.